src/v/kafka/client/BUILD

```diff
@@ -97,3 +97,28 @@ redpanda_cc_library(
         "@seastar",
     ],
 )
+
+redpanda_cc_library(
+    name = "record_batcher",
```

Comment:
question: why move it to k/client and not say storage?

Reply:
yeah fair. totally arbitrary at this point. more discussion [here](https://github.com/redpanda-data/redpanda/pull/23770#pullrequestreview-2367311314) if you want to chime in.

Reply:
in the interest of minimizing git shenanigans, i suggest we leave it here and then move it to `kafka/data` once https://github.com/redpanda-data/redpanda/pull/23770 merges.